TypeScript errors: `Property 'post' does not exist on type 'Agent'`

Upgrading from `4.3.0` to `5.1.1` broke our typechecking. Code like this

```ts
chai.request
  .agent(this._server)
  .post('/api/preview')
```

now fails with this error:

```
error TS2339: Property 'post' does not exist on type 'Agent'.
```

We think it's because `@types/superagent` is now in `devDependencies`, instead of `dependencies`. I think it should be in `dependencies`, because your package exports `superagent` types publicly. Since the superagent types are part of your public types, any package which uses your types should also include the `superagent` types that you export.

_Originally posted by @jthemphill in https://github.com/chaijs/chai-http/pull/310#discussion_r1833573576_

--- a/package.json
+++ b/package.json
@@ -57,12 +57,12 @@
     "is-ip": "^5.0.1",
     "methods": "^1.1.2",
     "qs": "^6.12.1",
+    "@types/superagent": "^8.1.7",
     "superagent": "^9"
   },
   "devDependencies": {
     "@eslint/js": "^9.3.0",
     "@types/chai": "^4.3.16",
-    "@types/superagent": "^8.1.7",
     "chai": "^5.1.0",
     "coveralls": "^3.1.1",
     "eslint": "^9.3.0",

